### PR TITLE
H1_Russell-L100.population_downscale_ctry

### PR DESCRIPTION
**Pull request by russellhz**
#402 
Replaced with a for loop and removed all hardcoding.
Also changed the next line to avoid spreading and regathering data.

### CODE REVIEW

**Comment by bpbond** [comment]
Thanks @russellhz . Note that runtime of `L100.Population_downscale_ctry` is unchanged by this PR.

File: R/zchunk_L100.Population_downscale_ctry.R [hunk 1]
```diff
@@ -101,45 +102,47 @@ module_socioeconomics_L100.Population_downscale_ctry <- function(command, ...) {
...
-      gather(year, pop_ratio_1950, -iso) %>%
-      mutate(year = as.integer(substr(year, 2, 5)))
     # Need to create matching iso codes for three countries in UN, but not Maddison.
     # Set Serbia and Montenegro ratio equal to Serbia & Montenegro and use Indonesia population ratio for East Timor
     mne_srb_tls <- filter(maddison_hist_ratio, iso %in% c("idn", "scg")) %>%
-      spread(iso, pop_ratio_1950) %>%
-      rename(mne = scg, tls = idn) %>%
-      mutate(srb = mne) %>%
-      gather(iso, pop_ratio_1950, -year)
+      mutate(iso = replace(iso, iso == "idn","tls"), iso = replace(iso, iso == "scg", "mne"))
```
Comment (bpbond) [inline R/zchunk_L100.Population_downscale_ctry.R]: Use `change_iso_code`

File: R/zchunk_L100.Population_downscale_ctry.R [hunk 2]
```diff
@@ -155,7 +158,7 @@ module_socioeconomics_L100.Population_downscale_ctry <- function(command, ...) {
       select(-year) %>%
       full_join(maddison_hist_ratio, by = "iso") %>%
       complete(year, nesting(iso)) %>%  # Completes tibble to include all years for all iso (creates missing values)
-      mutate(pop = if_else(!is.na(pop_ratio_1950), (pop * pop_ratio_1950), 0)) %>% # Set the remaining mismatched (very small) countries to zero
+      mutate(pop = if_else(!is.na(pop_ratio), (pop * pop_ratio), 0)) %>% # Set the remaining mismatched (very small) countries to zero
```
Comment (bpbond) [inline R/zchunk_L100.Population_downscale_ctry.R]: Reordering these clauses to `if_else(is.na(), ..., ...)` might be clearer.

File: R/zchunk_L100.Population_downscale_ctry.R [hunk 3]
```diff
@@ -101,45 +102,47 @@ module_socioeconomics_L100.Population_downscale_ctry <- function(command, ...) {
...
+      filter(hist_interp, iso != "world_total") %>%
+      left_join(pop_missing, by = "year")
+
+    maddison_hist_years <- unique(maddison_hist_ratio$year)
+
+    # Fill in missing values for period t using ratio of country to total population in period t+1 times the missing share in period t.
+    for( i in rev(maddison_hist_years)[-1]){
+      maddison_hist_ratio %>%
+        ungroup() %>%
+        mutate(pop2 = if_else(is.na(pop), (lead(pop, n = 1L, order_by = iso)), 0)) %>%
```
Comment (d3y419) [inline R/zchunk_L100.Population_downscale_ctry.R]: i know this is my fault. but can we make this better? 
i think pop2 should be previous_pop/
and else should probably be NA. 
0 and pmax was a hack.

Comment (bpbond) [inline R/zchunk_L100.Population_downscale_ctry.R]: It seems like this is an algorithm change that should be entered as issue and saved for V2?

Comment (d3y419) [inline R/zchunk_L100.Population_downscale_ctry.R]: russell talked me out of it. apparently, this "hack" is indeed the most efficient method. 
my other suggestion involves another if_else statement, with no obvious gain.

**Comment by bpbond** [comment]
Thanks for dealing with this @russellhz

File: R/zchunk_L100.Population_downscale_ctry.R (hunk 1 above)
Comment (d3y419) [inline R/zchunk_L100.Population_downscale_ctry.R]: i'm not entirely sure what happened here. is this resolved?

**Pull request merged**